components/automate-cli/cmd/chef-automate/ha_node_add_onprem.go

```diff
@@ -190,13 +190,47 @@ func (ani *AddNodeOnPremImpl) promptUserConfirmation() (bool, error) {
 	return ani.writer.Confirm("This will add the new nodes to your existing setup. It might take a while. Are you sure you want to continue?")
 }
 
+// Save all config from each services to Bastion server and move it to WORKSPACE dir
+func (ani *AddNodeOnPremImpl) saveConfigToBastion(infra *AutomateHAInfraDetails) error {
+	nodeObjects := getNodeObjectsToFetchConfigFromAllNodeTypes()
+	return executeCmdInAllNodeAndCaptureOutput(nodeObjects, true, AUTOMATE_HA_AUTOMATE_NODE_CONFIG_DIR, infra, ani.nodeUtils)
+}
+
+// Sync saved config from bastion to ALL nodes
+func (ani *AddNodeOnPremImpl) syncConfigToAllNodes(infra *AutomateHAInfraDetails) error {
+	nodeObjects := getNodeObjectsToPatchWorkspaceConfigToAllNodes()
+	return executeCmdInAllNodeAndCaptureOutput(nodeObjects, false, "", infra, ani.nodeUtils)
+}
+
 func (ani *AddNodeOnPremImpl) runDeploy() error {
-	err := ani.nodeUtils.writeHAConfigFiles(existingNodesA2harbTemplate, ani.config)
+	infra, _, err := ani.nodeUtils.getHaInfraDetails()
 	if err != nil {
 		return err
 	}
+	err = ani.saveConfigToBastion(infra)
+	if err != nil {
+		return errors.Wrap(err, "error saving node configuration to bastion")
+	}
+	err = ani.nodeUtils.writeHAConfigFiles(existingNodesA2harbTemplate, ani.config)
+	if err != nil {
+		return errors.Wrap(err, "error writing HA config.toml in workspace directory")
+	}
 	argsdeploy := []string{"-y"}
-	return ani.nodeUtils.executeAutomateClusterCtlCommandAsync("deploy", argsdeploy, upgradeHaHelpDoc)
+	err = ani.nodeUtils.executeAutomateClusterCtlCommandAsync("deploy", argsdeploy, upgradeHaHelpDoc)
+	if err != nil {
+		err = errors.Wrap(err, "error while deploying architecture")
+	}
+
+	infra, _, _ = ani.nodeUtils.getHaInfraDetails()
+	syncErr := ani.syncConfigToAllNodes(infra)
+	if syncErr != nil {
+		err = errors.Wrapf(syncErr, "error syncing config to all nodes. \n%v", err)
+	}
+
+	if err != nil {
+		return err
```

Comment:
can return directly above no need to check the error and assign it to new variable and then re-check and return